pylint/config/argument.py

```diff
@@ -80,6 +81,14 @@ def _non_empty_string_transformer(value: str) -> str:
     return pylint_utils._unquote(value)
 
 
+def _path_transformer(value: str) -> str:
+    """Check that a path is valid and expand user / variable."""
+    # Expand tilde to allow spelling-private-dict-file = ~/.pylintdict
+    value = os.path.expanduser(value)
+    value = os.path.expandvars(value)
+    return value
```

Comment:
```suggestion
    """Expand user and variables in a path."""
    return os.path.expandvars(os.path.expanduser(value))
```

Let's keep it simple. We also don't really check if the path is valid anymore.